src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPathContextProvider.cs

```diff
@@ -169,7 +169,8 @@ public bool TryCreateContext(string projectUniqueName, out IVsPathContext output
             catch (Exception exception)
             {
                 _telemetryProvider.PostFault(exception, typeof(VsPathContextProvider).FullName);
-                throw;
+                outputPathContext = null;
```

Comment:
We're still posting a fault so we'd still get noticed in the tracking. 

I think we should remove the fault post. 

@zivkan any thoughts on whether failure results are any useful here?

Reply:
We should remove the fault reporting only from targeted exception handlers, not the generic `catch (Exception e)` handler.  If we know there is a scenario that is causing a large number of fault telemetry, and we think it's a scenario that the customer should handle, not our own code, then we need to find a way to catch it in a non-generic exception handler.

The point is, if we add a regression in the future (for example, side-effect from more refactoring of VsSolutionManager, or CpsNuGetProject), we want to get fault telemetry.

Reply:
So you're suggesting we try to differentiate between an `expected` failure and one that isn't expected?

So catch InvalidOperationException or whatever the exception for a project not nominated does?

Reply:
I think Nikolche suggestion is really good. 

So, because a missing assets file is an error the customer should handle, we can remove the fault telemetry and remove the throw. And if the exception is a different than expected we should report the fault telemetry.

Reply:
> So you're suggesting we try to differentiate between an expected failure and one that isn't expected?

Yes, exactly. This is precisely why last year [I added the `ProjectNotNominatedException`](https://github.com/NuGet/NuGet.Client/commit/0fd1b7bd4c2643a4f61475875a7a4a4c40bc6f6f#diff-c70ce0d774938c0004954c3a8c426dfb472d5b497e24509b189d707c68f04bf4). I created this specific exception which is only used in a well-known scenario, so I can tell the difference between this and other unexpected `InvalidOperationException`.

I know some people aren't a fan of having hundreds of different exception types, but personally I think it's more important (see also [this issue](https://github.com/NuGet/Home/issues/12032) where NuGet.Protocol.dll using `FatalProtocolException` for many, many different reasons, without being able to programatically distinguish between the different scenarios, makes it harder for us to debug a customer issue).